shell/geta.go

```diff
@@ -0,0 +1,185 @@
+package shell
+
+import (
+	"errors"
+	"fmt"
+	"time"
+	"os/exec"
+	"strings"
+	"archive/zip"
+	"os"
+	"io"
+	"github.com/abiosoft/ishell"
+	"path/filepath"
+	"github.com/juruen/rmapi/model"
+)
+
+
+func getaCmd(ctx *ShellCtxt) *ishell.Cmd {
+	return &ishell.Cmd{
+		Name:      "geta",
+		Help:      "copy remote file to local with annotations. Optional [svg, pdf]",
+		Completer: createEntryCompleter(ctx),
+		Func: func(c *ishell.Context) {
+			// Parse cmd args
+			if len(c.Args) == 0 {
+				c.Err(errors.New("missing source file"))
+				return
+			}
+			srcName := c.Args[0]
+
+			// Download document as zip
+			node, err := ctx.api.Filetree.NodeByPath(srcName, ctx.node)
+			if err != nil || node.IsDirectory() {
+				c.Err(errors.New("file doesn't exist"))
+				return
+			}
+
+			err = getAnnotatedDocument(ctx, node, "")
+			if err != nil {
+				c.Err(err)
+				return
+			}
+		},
+	}
+}
+
+
+func getAnnotatedDocument(ctx *ShellCtxt, node *model.Node, path string) error {
+	zipFile := fmt.Sprintf(".%s.zip", node.Name())
+
+	// Set output name and output file name
+	output := node.Name()
+	if(path != ""){
+		output = fmt.Sprintf("%s/%s", path, node.Name())
+	}
+	outputFileName := fmt.Sprintf("%s.pdf", output)
+
+	// Parse last modified time of document on api
+	modifiedClientTime, err := time.Parse(time.RFC3339Nano, node.Document.ModifiedClient)
+	if err != nil {
+		// If we could not parse the time correctly, we still continue 
+		// with the execution such that the pdf is downloaded...
+		fmt.Println(err)
+		fmt.Println("(Warning) Could not parse modified time. Overwrite existing file.")
+		modifiedClientTime = time.Now().Local()
+	}
+
+	// If document has not changed since last update skip pdf convertion
+	outputFile, err := os.Stat(outputFileName)
+	if !os.IsNotExist(err) {
+		outputFileModTime := outputFile.ModTime()
+		if(outputFileModTime.Equal(modifiedClientTime)){
+			fmt.Println("Nothing changed since last download. Skip. ")
```

Comment:
Same here. This looks like `log.Trace` candidate.

Reply:
Thanks for your comments => fixed all invalid log statements